server/schedulers/hot_region.go

```diff
@@ -229,27 +259,35 @@ func (h *balanceHotRegionsScheduler) balanceHotReadRegions(cluster opt.Cluster)
 const balanceHotRetryLimit = 10
 
 func (h *balanceHotRegionsScheduler) balanceHotWriteRegions(cluster opt.Cluster) []*operator.Operator {
+	balancePeer := h.r.Int()%2 == 0
 	for i := 0; i < balanceHotRetryLimit; i++ {
-		if h.allowBalanceRegion(cluster) && (!h.allowBalanceLeader(cluster) || h.r.Int()%2 == 0) {
+		balancePeer = !balancePeer
```

Comment:
I'm a little confused about the assignment. I think you can directly use `!` or choose a different variable name.

Reply:
This is flipping `balancePeer` in each iteration. So, it will try `balanceByPeer` once then `balanceByLeader` then `balanceByPeer` again and so on.

Reply:
got it